sdk/src/main/java/com/bugsnag/android/Stacktrace.java

```diff
@@ -21,7 +24,8 @@
     public void toStream(@NonNull JsonStream writer) throws IOException {
         writer.beginArray();
 
-        for (StackTraceElement el : stacktrace) {
+        for (int k = 0; k < stacktrace.length && k < STACKTRACE_TRIM_LENGTH; k++) {
```

Comment:
my understanding of this change is that it omits anything beyond the first 200 values in that stacktrace when serializing. How does it work if the report fails to send? Does the serialized report get saved to storage or does the original report entity get saved? What I'm trying to get at here is whether an unruly stacktrace will be reallocated some memory and/or persistent storage… or is it just passed in by reference and always operated on in its initially allocated memory?

Reply:
If a report fails to send it will be persisted to disk as a file using the same `toStream` method. That file will then be resent without any knowledge of its contents.